docs/Index.tsx

```diff
@@ -287,6 +287,14 @@ class App extends React.Component<{}, AppState> {
         document.addEventListener(SideNavigation.toggleEvent, () => {
             this.setState({sideNavOpened: !this.state.sideNavOpened});
         });
+
+        window.onhashchange = () => {
+            const componentIdFromHash = this.getHash();
+            const firstComponentId = this.components.sort(this.sortComponentsByName);
+            this.setState({
+                activeComponentId: this.getSelectedComponent(componentIdFromHash) && componentIdFromHash || firstComponentId[0].componentName,
+            });

```

Comment:
You're pretty much duplicating the constructor, you should probably extract a method to get the componentId from the url

Reply:
changed